src/permission/rule-parser.js

```diff
@@ -0,0 +1,109 @@
+var rulesMap = require( './rules-map' );
+var FUNCTION_REGEXP = /([a-zA-Z0-9_]*)\(/g;
```

Comment:
var FUNCTION_REGEXP = /([\w]*)\(/g;
var NEW_REGEXP = /(^|[^\w~])new[^\w~]/;
var OLD_DATA_REGEXP = /(^|[^\w~])oldData[^\w~]/;
var DATA_REGEXP = /(^|[^\w~])data[^\w~]/;
var CROSS_REFERENCE_REGEXP = /(^|[^\w~])_[^\w~]/;

This lets tests pass, reduces complexity and also fixes a bug where '_data and _oldData' was considered legit